isis/src/base/apps/isisimport/isisimport.cpp

```diff
@@ -22,33 +23,41 @@ namespace Isis {
 
   void isisimport(UserInterface &ui, Pvl *log) {
     FileName fileTemplate = ("$ISISROOT/appdata/import/fileTemplate.tpl");
-    FileName xmlFileName = ui.GetFileName("FROM");
+    json jsonData;
+    bool isPDS4 = false;
+    FileName inputFileName = ui.GetFileName("FROM");
 
-    // To read the DN data
-    ProcessImport importer;
-    if (xmlFileName.removeExtension().addExtension("dat").fileExists()){
-      importer.SetInputFile(xmlFileName.removeExtension().addExtension("dat").expanded());
+    if (inputFileName.extension().toUpper() == "IMQ"){
+      QString msg = "Input image may be compressed. Please run image through vdcomp to uncompress"
+                    "or verify image has correct file extension.";
+      throw IException(IException::User, msg, _FILEINFO_);
     }
-    else if (xmlFileName.removeExtension().addExtension("img").fileExists()) {
-      importer.SetInputFile(xmlFileName.removeExtension().addExtension("img").expanded());
+
+    try {
+      // try to convert xml file to json
+      jsonData = xmlToJson(inputFileName.toString());
+      isPDS4 = true;
     }
-    else {
-      QString msg = "Cannot find image file for [" + xmlFileName.name() + "]. Confirm that the "
-        ".dat or .img file for this XML exists and is located in the same directory.";
-      throw IException(IException::User, msg, _FILEINFO_);
+    catch(...) {
+      try {
+        // try to convert pvl to json
+        jsonData = pvlToJSON(inputFileName.toString());
+      }
+      catch(...) {
+        QString msg = "Unable to process import image. Please confirm image is in PDS3 or PDS4 format";
+        throw IException(IException::User, msg, _FILEINFO_);
+      }
     }
```

Comment:
I want to say there is a better way to do this but I am not sure. Ideally we could enumerate our drivers, current that is only pvlToJson and xmlToJson, the just loop over and try each of them. ISIS has typically done this with factories but I think we could potentially just operate using a list of functions

Reply:
I agree. I feel there is a better way as well, but was unsure how to keep a tracker for whether or not the format was an XML. This is due to the fact that if it's XML , we need to add the OriginalXmlLabel to the cube.

Reply:
Agreed, let's add this as a todo/think about